### PR TITLE
fix: update markdown text to refresh on locale change

### PR DESCRIPTION
**Pull request by hanlunBCRegistries**
Issue:
#281 
### Summary

This pull request fixes the issue where markdown text does not update when changing the locale between English and French.
### Changes

- Added a `watch` function to observe locale changes and refresh the content dynamically.
- Ensured that content updates without requiring a page refresh.

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-ar-dev--pr-282-91dl1dvn.web.app

File: web/site/app.vue [hunk 1]
```diff
@@ -17,15 +17,21 @@ useHead({
 
 const appVersion = await getAppMetaInfo() // load ui and api version on app mount
 
-// query help markdown and globally provide to use in either pages/help.vue or <SbcHelpModal />
-// if queried directly from modal, it wont be prerendered because modal is hidden initially
-const { data: helpDocs } = await useAsyncData('help-docs-query', () => {
+// Query help markdown and globally provide it for use in either pages/help.vue or <SbcHelpModal />
+const { data: helpDocs, refresh: refreshHelpDocs } = await useAsyncData('help-docs-query', () => {
```
Comment (deetz99) [inline web/site/app.vue:21]: Can you not use the watch option provided by useAsyncData?

https://nuxt.com/docs/api/composables/use-async-data#watch-params

File: web/site/components/Sbc/NuxtContentCard.vue [hunk 2]
```diff
@@ -18,11 +18,16 @@ const props = defineProps({
 
 const fullId = 'content-data-' + props.id
 
-const { data } = await useAsyncData(fullId, () => {
+// Fetch the content data based on the current locale and route
+const { data, refresh } = await useAsyncData(fullId, () => {
```
Comment (deetz99) [inline web/site/components/Sbc/NuxtContentCard.vue:22]: Can you not use the watch option provided by useAsyncData?

https://nuxt.com/docs/api/composables/use-async-data#watch-params

Comment (hanlunBCRegistries) [reply web/site/components/Sbc/NuxtContentCard.vue]: Sure, just updated

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-ar-dev--pr-282-dfzms25r.web.app

**Comment by EPortman** [comment]
If I switch the language in the header to French:
image

The page ends up completely blank.
image

Until I refresh the page at which point it updates.
image


Is that supposed to be the case? I am using the /gcbrun url.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-ar-dev--pr-282-dfzms25r.web.app

**Comment by EPortman** [comment]
Never mind. It seems to be working now.

**Review by EPortman — APPROVED** [review]
Working on my end. Cannot say anything about the French though.

**Pull request merged**